src/client/api/init.c

```diff
@@ -157,11 +157,21 @@ daos_init(void)
 	if (rc != 0)
 		D_GOTO(out_debug, rc);
 
+	/** set up agent */
+	rc = dc_agent_init();
+	if (rc != 0)
+		D_GOTO(out_hhash, rc);
+
+	/** get CaRT configuration */
+	rc = dc_network_cfg(NULL);
+	if (rc != 0)
+		D_GOTO(out_obj, rc);
```

Comment:
Shouldn't this be jumping to `out_hhash`?

Reply:
If dc_network_cfg() fails, we want it to clean up from dc_agent_init().  That function is cleaned up by out_obj.  Jumping to out_obj is correct.  You may observe that daos_eq_lib_init() also jumps to out_obj on failure.  It jumps to the same location as dc_network_cfg() because dc_network_cfg() cleans up after itself and doesn't have/need its own cleanup routine.